### ISSUE TITLE
Enum value is not properly cleaned

### ISSUE THREAD

**Issue by philidem** [philidem/typed-model#4]
**Example:**
```javascript
var AddressType = Enum.create({
    values: {
        'home': {
            title: 'Home'
        },

        'work': {
            title: 'Work'
        }
    }
});

var Address = Model.extend({
    properties: {
        city: String,
        state: String,
        type: AddressType
    }
});

var address = new Address({
    city: 'Durham',
    state: 'NC',
    type: AddressType.WORK
});

// This expectation is currently failing
expect(Model.clean(address)).to.deep.equal({
    city: 'Durham',
    state: 'NC',
    type: 'work'
});
```

**Comment by philidem** [comment]
@austinkelleher, I have a fix for this.

**Comment by philidem** [comment]
Fixed in `typed-model@3.0.2`
